### DISCUSSION TITLE
Shortest path problem

### DISCUSSION THREAD

**Discussion by FlorBel** [pysal/momepy#511] [Q&A] [answered]
I am struggling with a momepy-problem, and I have two questions.  
I am working on a shortest distance problem as I want to connect a certain source of heat to a certain sink (cluster of houses). 

The network I am using is a street network shapefile and all the streets have a certain ID and cost of distribution. 
Both files from sources and sinks are point data, but I can connect them to the main streetnetwork, by creating a line with a certain name for instance.

When I use momepy’s gdf_to_nx I don’t have a clue how the streets are named and I seem not to be able to find a certain street or line, and that is very important to make a calculation from source to sink.
The network is too extensive to do a calculation for the shortest distance on all nodes and then perform a nx_to_gdf.  
 
Secondly, as I mentioned, I have a cost per street segment what is would cost to create a heat distribution network, is it  
possible to keep the value of the attribute/field in the graph? So that it can be used as a weight in the shortest distance? 
In that way I can look for the cheapest trajectory rather than just shortest or fastest.

**Comment by martinfleis** [comment, marked as answer]
All attributes shall be preserved in the Graph, including ID and the cost. Those will be stored as [edge](https://networkx.org/documentation/stable/reference/classes/generated/networkx.Graph.edges.html) attributes and be directly used as weight in a computation. I believe this answers both questions.
